### PR TITLE
todo for improving logging for non-existing objects / handle standard…

### PR DESCRIPTION
**Pull request by codesurf42**
… aws exceptions

### CODE REVIEW

**Comment by kastoestoramadus** [comment]
@codesurf42 Create a Jira ticket for that.

File: app/uk/gov/hmrc/fileupload/controllers/FileDownloadController.scala [hunk 1]
```diff
@@ -34,16 +35,22 @@ class FileDownloadController(
...
+        Result(
+          header = ResponseHeader(200, Map.empty),
+          body = HttpEntity.Streamed(
+            result.stream,
+            Some(result.metadata.contentLength),
+            Some(result.metadata.contentType))
+        )
+      case None => Result(
+        header = ResponseHeader(404, Map.empty),
+        body = HttpEntity.Strict(ByteString("{msg:'File not fould'}"), Some("application/json"))
```
Comment (kastoestoramadus) [inline app/uk/gov/hmrc/fileupload/controllers/FileDownloadController.scala]: I am not sure of this body.

File: app/uk/gov/hmrc/fileupload/controllers/FileDownloadController.scala [hunk 2]
```diff
@@ -34,16 +35,22 @@ class FileDownloadController(
...
+        Result(
+          header = ResponseHeader(200, Map.empty),
+          body = HttpEntity.Streamed(
+            result.stream,
+            Some(result.metadata.contentLength),
+            Some(result.metadata.contentType))
+        )
+      case None => Result(
+        header = ResponseHeader(404, Map.empty),
+        body = HttpEntity.Strict(ByteString("{msg:'File not found'}"), Some("application/json"))
```
Comment (kastoestoramadus) [inline app/uk/gov/hmrc/fileupload/controllers/FileDownloadController.scala]: @craigtd What would be proper body response?

Comment (codesurf42) [inline app/uk/gov/hmrc/fileupload/controllers/FileDownloadController.scala]: Is there any hmrc-standard/canonical way of error response? The only reason for body is to carry the more detailed message

File: app/uk/gov/hmrc/fileupload/testonly/S3TestController.scala [hunk 3]
```diff
@@ -86,10 +86,10 @@ trait S3TestController { self: Controller =>
   def s3downloadFileT(fileName: String, version: Option[String]) = s3downloadFile(transientBucketName, fileName, version)
 
   def s3downloadFile(bucket: String, fileName: String, version: Option[String]) = Action { req =>
-    Logger.info(s"downloading $fileName from bucket: $bucket")
+    Logger.info(s"downloading $fileName from bucket: $bucket, versionO: $version")
```
Comment (codesurf42) [inline app/uk/gov/hmrc/fileupload/testonly/S3TestController.scala]: versionO?

File: app/uk/gov/hmrc/fileupload/testonly/S3TestController.scala [hunk 4]
```diff
@@ -86,10 +86,10 @@ trait S3TestController { self: Controller =>
...
     val result = version match {
       case Some(v) => s3Service.download(bucket, S3KeyName(fileName), v)
-      case None => s3Service.download(bucket, S3KeyName(fileName))
+      case None => s3Service.download(bucket, S3KeyName(fileName)).get
```
Comment (codesurf42) [inline app/uk/gov/hmrc/fileupload/testonly/S3TestController.scala]: would be more consistent to update both methods to options

File: app/uk/gov/hmrc/fileupload/testonly/S3TestController.scala (hunk 3 above)
Comment (kastoestoramadus) [inline app/uk/gov/hmrc/fileupload/testonly/S3TestController.scala]: version Option

**Comment by kastoestoramadus** [comment]
@codesurf42 Are you happy to merge it?

**Comment by kastoestoramadus** [comment]
@codesurf42 ready

**Comment by kastoestoramadus** [comment]
I would like to merge it. Any objections?

**Pull request merged**